.travis.yml

```diff
@@ -7,9 +7,27 @@ services:
 
 matrix:
   include:
+    # Nightly builds run the tests/examples end-to-end test suite.
+    # Does not include Windows since Projects/Models are mostly not compatible with Windows as of now.
+    - stage: Nightly
+    - language: python
+      python: 2.7
+      install:
+        # CHANGED_FILES detects whether or not files related to the nightly build are changed.
+        # True is so that the command returns a 0 exit code whether or not CHANGED_FILES is empty.
+        - CHANGED_FILES=$(git diff --name-only master..HEAD | grep "tests/examples\|examples") || true;
```

Comment:
Question, was there a reason not to use `TRAVIS_COMMIT_RANGE`? Sometimes we merge PRs into branches other than master (e.g. release branches like `branch-1.4`), so it might be safest to not assume that we should diff `master` & `HEAD` & use `TRAVIS_COMMIT_RANGE` instead.

Reply:
Makes sense, though will have to use `${TRAVIS_COMMIT_RANGE/.../..}` instead according to [this Travis issue](https://github.com/travis-ci/travis-ci/issues/4596).

Reply:
> There's also this issue posted by someone in the issue:
> Note: there's a further underlying problem here that crops up for long-lived feature branches not rebased onto master. Travis uses a shallow clone with a default depth of 50. If the shallow clone doesn't include the originating commit where the feature branch diverged, then `$TRAVIS_COMMIT_RANGE` will always include the full repository ancestry of the feature branch going back to the first commit. Both double dot and triple dot commit ranges are predicated on having the full repository ancestry.

Do you think this would be an issue regarding our repository?

Reply:
After offline discussion, we decided that using `master..HEAD` was ok for now since PRs for release branches are rare and it wouldn't hurt to run nightly tests for them.